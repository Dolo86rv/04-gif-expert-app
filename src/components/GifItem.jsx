import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({title, id, url}) => {
    
    return (
    <div className="card">
        <p>ID. {id}</p>
        <img src={url} alt={title} />
        <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
