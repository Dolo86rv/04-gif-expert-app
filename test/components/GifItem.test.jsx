import { GifItem } from '../../src/components/GifItem'
import { render, screen } from '@testing-library/react'

describe('Pruebas en <GifItem />', () => { 
    
    const title = 'Hi'
    const url = 'https://api.com/'

    test('debe hacer match con el snapshot', () => { 
    
        const { container } = render(<GifItem  title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar el url y el Alt indicado', () => { 
        render( <GifItem title={title} url={url} />)
        //screen.debug()
        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url )
        expect( alt ).toBe( title )
     })

     test('debe mostrar el titulo como un String', () => { 
        render( <GifItem title={title} url={url} />)

        expect( screen.getByText(title)).toBeTruthy()
      })
 })