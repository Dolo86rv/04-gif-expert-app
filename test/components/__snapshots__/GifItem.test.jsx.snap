// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      ID. 
    </p>
    <img
      alt="Hi"
      src="https://api.com/"
    />
    <p>
      Hi
    </p>
  </div>
</div>
`;
